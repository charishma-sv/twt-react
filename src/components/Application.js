import React, { useContext } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import HomePage from './HomePage';
import Login from './Login';
import Feed from './Feed';
import { UserContext } from '../providers/UserProvider';
import ProtectedRoute from './Routes/ProtectedRoute';
import PublicRoute from './Routes/PublicRoute';
import Signup from './Signup';
const Application = () => {
  const { user, updateUser } = useContext(UserContext);
  return (
    <Router>
      <div>
        <PublicRoute component={Login} user={user} path="/login" />
        <PublicRoute
          component={Signup}
          user={user}
          updateUser={updateUser}
          path="/signup"
        />
        <ProtectedRoute component={Feed} user={user} path="/feed" />
        <PublicRoute component={HomePage} user={user} path="/" exact />
      </div>
    </Router>
  );
};
export default Application;
