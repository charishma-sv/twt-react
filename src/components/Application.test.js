import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import Application from './Application';
describe('<Application />', () => {
  beforeEach(() => {
    render(<Application />);
  });
  describe('Application snapshot ', () => {
    it('should render application snapshot', () => {
      let tree = create(<Application />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
