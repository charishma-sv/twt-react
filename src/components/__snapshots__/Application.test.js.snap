// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Application /> Application snapshot  should render application snapshot 1`] = `
<div>
  <div
    className=" vh-100 d-flex justify-content-center align-items-center container"
  >
    <div
      className="spinner-grow"
    />
  </div>
</div>
`;
